### PR TITLE
Add Data documentation page

### LINKED ISSUE #3707: Create Data page
- [x] Create "Data" page on Design System site
- [x] Include intro sentence/paragraph on what the component family is (could borrow from release notes sentence)
- [x] Include short section on the subcomponents/their purpose (this can be a 1 sentence per subcomponent. eg. "DataSort allows user to sort by a specific property.")

Deliverable:
- [ ] PR merged to site

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-3706--keen-mayer-a86c8b.netlify.app/components/data)

#### What does this PR do?

Adds "Data" page which gives an example, speaks lightly to its subcomponents and mentions 2 ways to build with it. The "how to" guides referenced will be added as separate PRs. Although, we already have one published so it feels reasonable to include the link now.

- [x] Need to replace the preview thumbnail

Closes #3707 

#### Where should the reviewer start?

data.mdx

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

### CODE REVIEW

File: aries-site/src/pages/components/data.mdx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  showResponsiveControls={false}
+>
+  <DataExample />
+</Example>
+
+## Subcomponents
+
+### DataFilter
+
+An individual filter for a property. Based on the type of data for that property, DataFilter determines a suitable "smart default" for what kind of input to present.
```
Comment (britt6612) [inline aries-site/src/pages/components/data.mdx:31]: wonder if we can mention that the "smart default" is picked based on the chart below.

**Review by britt6612 — COMMENTED** [review]
going to review the code next but the content looks good as a starting point

File: aries-site/src/data/structures/components.js [hunk 2]
```diff
@@ -1746,6 +1746,27 @@ export const components = [
       'Cover',
     ],
   },
+  {
+    name: 'Data',
+    available: true,
+    category: 'All',
```
[the remark this answers is not in the document]
Comment (taysea) [reply aries-site/src/data/structures/components.js:1752]: Good call, yeah I'm supportive of making it its own section

[the remark this answers is not in the document]
Comment (britt6612) [reply aries-site/src/data/structures/components.js:1752]: okay same to align with Grommet

File: aries-site/src/pages/components/data.mdx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+Data is the wrapper component and orchestrator for a set of subcomponents. Because this set of components work together to create the entirety of the experience, we like to refer to them as "Data and friends".
+
+<Example
+  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/data/DataSearchExample.js"
+  docs="https://v2.grommet.io/data"
+  template
+  height="auto"
+  showResponsiveControls={false}
+>
+  <DataExample />
```
Comment (jcfilben) [inline aries-site/src/pages/components/data.mdx:24]: The pagination for this example says "showing 1-10 of 21 items", but more than 10 items are visible on the first page.

Screen Shot 2024-02-09 at 1 25 55 PM

Comment (taysea) [reply aries-site/src/pages/components/data.mdx:24]: This is fixed now. It seems to be something with when `views` is defined. I've removed it for now and will file a grommet ticket to investigate separately.

Comment (jcfilben) [reply aries-site/src/pages/components/data.mdx:24]: Another similar issue, when I search and end up with 2 results the pagination shows that it is showing '1-10' items. We can look into separately from this PR though.

Screen Shot 2024-02-12 at 3 41 21 PM

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by britt6612 — APPROVED** [review]
LGTM

**Pull request merged**